src/table/src/EditableCell.js

```diff
@@ -47,40 +103,76 @@ class EditableCell extends React.PureComponent {
   }
 
   handleDoubleClick = () => {
-    this.setState({
-      isEditing: true
-    })
+    if (this.props.disabled || !this.props.isSelectable) return
+    if (this.isControlled()) {
+      this.props.onEditStart()
+    } else {
+      this.setState({
+        isEditing: true
+      })
+    }
   }
 
   handleKeyDown = e => {
+    if (this.props.disabled) return
     const { key } = e
+
+    /**
+     * When the user presses a character on the keyboard, use that character
+     * as the value in the text field.
+     */
     if (key.match(/^[a-z]{0,10}$/) && !e.metaKey && !e.ctrlKey && !e.altKey) {
```

Comment:
This can be a bit confusing! `EditableCell` is actually just a selectable `Table.TextCell`. As soon as you hit enter, space or a character — a component called `EditableCellField` which contains a `Textarea` is overlayed exactly on top of the table cell. The `EditableCellField` position will be kept in sync. The reason for matching the key and passing the key as the value is then passed to the Textarea. This is inline with how spreadsheets work.